### PR TITLE
4489 router logger

### LINKED ISSUE #4489: Problem with Payload Logger set up for ROUTER model
In the helm chart for my `SeldonDeployment`, when I specify it in the parent ROUTER model , no logs are sent (code snippet 1)
I need to specify (code snippet 2) to get it to work

Is there a better way to configure the logger to be called for the entire graph instead of specifying the logger in each of the children?

Reason here is that when specifying inference graphs of increasing complexity, ideally, I'd like to be able to log payloads for request and response payload going into/out of the entire graph. Not individual components.

*Code snippet 1:*
```
graph:
      children: 
        - name: <model_A>
          type: MODEL
        - name: <model_B>
          type: MODEL
      name: <router_name>
      type: ROUTER
      logger:
            url: <url>
            mode: all
      endpoint:
        type: REST
```
*Code snippet 2:*
```
graph:
      children: 
        - name: <model_A>
          type: MODEL
          logger:
            url: <url>
            mode: all
        - name: <model_B>
          type: MODEL
          logger:
            url: <url>
            mode: all
      name: <router_name>
      type: ROUTER
      endpoint:
        type: REST
```

### PR DESCRIPTION
**Pull request by cliveseldon**
**What this PR does / why we need it**:

Allows logging at ROUTER elements in graph.

**Which issue(s) this PR fixes**:

Fixes #4489

### CODE REVIEW

File: executor/predictor/predictor_process.go [hunk 1]
```diff
@@ -329,7 +336,17 @@ func (p *PredictorProcess) predictChildren(node *v1.PredictiveUnit, msg payload.
 				return cmsgs[0], err
 			}
 		}
-		return p.aggregate(node, cmsgs, msg, puid)
+		amsg, err := p.aggregate(node, cmsgs, msg, puid)
```
Comment (RafalSkolasinski) [inline executor/predictor/predictor_process.go:339]: As `aggregate` also calls `p.logPayload` would't some messages be logged twice?

Comment (cliveseldon) [reply executor/predictor/predictor_process.go:339]: I think the idea here is you can log the payload after every transformation. Normally, each node in the graph only does one thing like route or combine. But its true you can have a node that does multiple things in which case you would get payloads logged after each operation. The code in `aggregate`only logs if the operation is run.

**Comment by seldondev** [comment]
@cliveseldon: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
notebooks | c84984be683d1ab7ba40f97feb1832121cc48733 | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/io-seldon-core-pr-4533-notebooks-wjv2d) | `/test notebooks`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@cliveseldon: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
notebooks | c84984be683d1ab7ba40f97feb1832121cc48733 | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/io-seldon-core-pr-4533-notebooks-wjv2d) | `/test notebooks`
integration | c84984be683d1ab7ba40f97feb1832121cc48733 | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-4533-integration-lx569) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@cliveseldon: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
notebooks | c84984be683d1ab7ba40f97feb1832121cc48733 | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/io-seldon-core-pr-4533-notebooks-wjv2d) | `/test notebooks`
integration | c84984be683d1ab7ba40f97feb1832121cc48733 | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-4533-integration-5blch) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@cliveseldon: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | c84984be683d1ab7ba40f97feb1832121cc48733 | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-4533-integration-5blch) | `/test integration`
notebooks | c84984be683d1ab7ba40f97feb1832121cc48733 | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/io-seldon-core-pr-4533-notebooks-bx87r) | `/test notebooks`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by adriangonz** [comment]
Changes look good!

If I'm not mistaken, the integration and notebook errors should already be fixed in `master`. So if you can rebase @cliveseldon , we can kick them off again.

**Comment by seldondev** [comment]
@cliveseldon: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
notebooks | c84984be683d1ab7ba40f97feb1832121cc48733 | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/io-seldon-core-pr-4533-notebooks-bx87r) | `/test notebooks`
integration | 8d634f8511bc0ca4f3a811fb4c28602630775d1d | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-4533-integration-tv2qn) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@cliveseldon: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
notebooks | 8d634f8511bc0ca4f3a811fb4c28602630775d1d | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/io-seldon-core-pr-4533-notebooks-4s4th) | `/test notebooks`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Pull request merged**